### ISSUE TITLE
caching AST parsing 

### ISSUE THREAD

**Issue by jonathanong** [benjamn/recast#107]
any chance of caching the parsing of the AST? i'm particularly interested in this for a watching build process as recast becomes unbearably slow with larger files. 

could be included (difficult imo with configurable esprima and options) or an external function. would also be cool if we could cache it to disk (serialize), but that's probably excessive

**Comment by jonathanong** [comment]
anyone have any ideas on this?

**Comment by stefanpenner** [comment]
honestly the parsing is a extremely small portion of the time, the visiting/transforms is really the majority of the time.

Although this would result in an improvement, I suspect it will be a minority.

Honestly the best bang for your buck would be to cache the input/output based on some file content checksum, or even better a tool that doesn't rebuild a file if the file didn't change.

**Comment by jonathanong** [comment]
yeah i already do that. i'm making multiple builds of the same file with different transforms, which is where this optimization comes in.

iirc parsing mocha.js takes ~1s, transforms about ~300ms each, and printing ~2s. add 50% w/ AST. totally small optimization.
